rust/get-ledger-blocks/src/main.rs

```diff
@@ -0,0 +1,70 @@
+use std::error::Error;
+use ic_agent::Agent;
+use ic_agent::agent::http_transport::ReqwestHttpReplicaV2Transport;
+use ic_agent::ic_types::Principal;
+use prost::Message;
+use structopt::StructOpt;
+use url::Url;
+use ledger::GetBlockRequest;
+use crate::ledger::get_blocks_response::GetBlocksContent;
+use crate::ledger::{Block, GetBlocksResponse};
+
+pub mod ledger {
+    include!("gen/ledger.rs");
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "get-ledger-blocks", about = "An example of how to get blocks from the ledger")]
+struct Opt {
+    #[structopt(short, long)]
+    url: Url,
+
+    #[structopt(short, long)]
+    ledger_id: String,
+
+    #[structopt(short, long)]
+    start: u64,
+
+    #[structopt(short = "L", long)]
+    length: u64
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    let opt = Opt::from_args();
+    let transport = ReqwestHttpReplicaV2Transport::create(opt.url).unwrap();
+    let agent = Agent::builder()
+        .with_transport(transport)
+        .build()
+        .unwrap();
+    let ledger_principal_id = Principal::from_text(opt.ledger_id).unwrap();
+
+    // create the request and encode it as protobuf message
+    let req = GetBlockRequest {
+        start: opt.start,
+        length: opt.length
+    }.encode_to_vec();
+
+    // call get_blocks_pb on the Ledger with the encoded request as argument
+    let res = agent.query(&ledger_principal_id, "get_blocks_pb")
+        .with_arg(req)
+        .call()
+        .await?;
+
+    // decode the response as protobuf message
+    let res = GetBlocksResponse::decode(res.as_slice()).unwrap().get_blocks_content.unwrap();
+
+    // check if the response is an error or the encoded blocks
+    let encoded_blocks = match res {
+        GetBlocksContent::Blocks(blocks) => blocks.blocks,
+        GetBlocksContent::Error(e) => panic!("Error fetching blocks {}", e),
```

Comment:
I'm afraid that this example is too simplistic to be actually useful. Most blocks are in the archive, so you really need to fetch the archive index first and based on that request blocks

Reply:
Should I wait for the fetch blocks API implementation and only then push this example? I don't think I can query a range include archives without it.

Reply:
Yeah, using the new endpoint would be best. But we do query the index manually in rosetta right now:
https://gitlab.com/dfinity-lab/public/ic/-/blob/master/rs/rosetta-api/src/ledger_client.rs#L1096

Also, there is an endpoint for querying a single block which behaves just like the new interface will:
https://gitlab.com/dfinity-lab/public/ic/-/blob/master/rs/rosetta-api/src/ledger_client.rs#L950-973

Although we were planning to deprecate both and switch to the one @roman-kashitsyn has been working on...
@roman-kashitsyn what do you think?